Consolidate documentation to fix setup complexity

Andr1yK opened the pull request:
## Summary
Fixed hosts file copy-paste issue and consolidated all setup instructions into single `README.md` based on mentor feedback about junior developers struggling with scattered documentation.

Andr1yK committed:
Consolidate documentation to fix setup complexity

Fixed hosts file copy-paste issue and consolidated all setup instructions into single README.md based on mentor feedback about junior developers struggling with scattered documentation.

- Replace README.md with comprehensive single-file setup guide
- Remove Setup.md to eliminate file jumping
- Fix hosts file example removing problematic '+' symbol
- Add Windows/WSL support and troubleshooting
- Include SSL setup instructions inline
- Add warning for dangerous Docker commands

vadim-ilchenko commented on README.md:
```diff
@@ -1,87 +1,279 @@
+# Huntd Test Assignment
 
-Anonymous job search in IT (public version).
+Anonymous job search in IT (test version for Mate Academy candidates).
```
```suggestion
Anonymous job search in IT (test version for Mate academy candidates).
```

vadim-ilchenko commented on README.md:
```diff
@@ -1,87 +1,279 @@
+
+### 2. NPM Token Setup
+
+You'll receive a private NPM token from Mate Academy. Configure it in three places:
```
```suggestion
You'll receive a private NPM token from Mate academy. Configure it in three places:
```

vadim-ilchenko commented on README.md:
```diff
@@ -1,87 +1,279 @@
-To run the project run following command in the root directory (INSERT NPM TOKEN INTO MAKEFILE BEFORE RUNNING IT):
+**Step 1: Local NPM configuration**
+- **Linux/macOS/WSL:** `~/.npmrc`
+- **Windows:** `C:\Users\<YourUsername>\.npmrc`
```
```suggestion
- **Windows:** `C:/Users/<YourUsername>/.npmrc`
```

are there really backslashes on windows or is it a typo?

upd: [looks like no issues](https://stackoverflow.com/questions/1589930/so-what-is-the-right-direction-of-the-paths-slash-or-under-windows)
<img width="734" height="239" alt="image" src="https://github.com/user-attachments/assets/f015f370-5c5f-4ff5-9ef9-1b8f1f0f2f57" />

Andr1yK replied:
good point, fixed

vadim-ilchenko commented on README.md:
```diff
@@ -1,87 +1,279 @@
+### 2. SSL Certificate Configuration
 
-- **Redis:**
+SSL certificates are automatically generated. To access via HTTPS:
```
```suggestion
SSL certificates are pre-generated and stored in ... (add path here). To access via HTTPS:
```

vadim-ilchenko commented on README.md:
```diff
@@ -1,87 +1,279 @@
-  In production Huntd project uses AWS Redis.
+2. **Trust the certificate:**
+   - **macOS:** [Add to Keychain](https://tosbourn.com/getting-os-x-to-trust-self-signed-ssl-certificates/)
+   - **Windows:** [Install certificate](https://community.spiceworks.com/how_to/1839-installing-self-signed-ca-certificate-in-windows)
```
perhaps users should first visit the website to download certificate

or we can ask them to add certificate directly from the nginx folder to `trusted`

vadim-ilchenko commented on README.md:
```diff
@@ -1,87 +1,279 @@
+- **REST endpoint:** http://localhost:4000/rest (OAuth callbacks only)
+- **Technology:** Node.js, TypeScript, Sequelize, Apollo GraphQL
+
+Useful GraphQL learning resources:
```
wow!

vadim-ilchenko commented on README.md:
```diff
@@ -1,87 +1,279 @@
+### Database
+- **Technology:** PostgreSQL 10.21
+- **Port:** 5432
+- **Credentials:** user: `dev`, password: `772184`, database: `huntd_development`
```
fire!

vadim-ilchenko commented on README.md:
````diff
@@ -1,87 +1,279 @@
+make front  # Access Frontend container
+```
+
+### Daily Workflow
+
+1. **Adding dependencies:** Always rebuild the service container after adding packages:
+   ```bash
+   make rebuild-hard s=api        # After adding API dependencies
+   make rebuild-hard s=frontend   # After adding Frontend dependencies
+   ```
````
do we need it in test assignment?

Andr1yK replied:
Added it just in case, sometimes we need to rebuild containers

vadim-ilchenko commented on README.md:
````diff
@@ -1,87 +1,279 @@
+   make rebuild-hard s=frontend   # After adding Frontend dependencies
+   ```
+
+2. **Database changes:** Run migrations inside the API container:
````
add note to **NOT** run migrations from outside of the docker

vadim-ilchenko commented on README.md:
```diff
@@ -1,87 +1,279 @@
+
+## Troubleshooting
+
+### Docker Issues on Windows
```
ensure database is up

vadim-ilchenko commented on README.md:
```diff
@@ -1,87 +1,279 @@
+
+If the project runs slowly:
+1. **Windows:** Ensure you're using WSL 2 and files are on Linux filesystem
+2. **macOS:** Close unnecessary applications to free up memory
```
🤣

vadim-ilchenko commented on nginx/README.md:
```diff
@@ -24,10 +24,10 @@ SSL certificates can be placed in ./local-ssl if needed for advanced configurati
     127.0.0.1       localhost
     127.0.0.1       localunixsocket.local
-    + 127.0.0.1     local.huntd.tech
+    127.0.0.1       local.huntd.tech
```
🔥

Andr1yK committed:
Fix capitalization: 'Mate Academy' → 'Mate academy'

Andr1yK committed:
Address PR review comments: fix paths, SSL instructions, and migration warning

Fixed Windows .npmrc path format, specified SSL certificate location, added migration container warning, and restored original two-step certificate trust process.
